Formula/b/basedpyright.rb

```diff
@@ -0,0 +1,27 @@
+require "language/node"
+
+class Basedpyright < Formula
+  desc "Pyright fork with various improvements and built-in pylance features"
+  homepage "https://github.com/DetachHead/basedpyright"
+  url "https://registry.npmjs.org/basedpyright/-/basedpyright-1.15.2.tgz"
+  sha256 "c9e0dcceb4c09c8a5426eb544c1c1991fd27415f313c12e594e1c3373bea6aa4"
+  license "MIT"
+  head "https://github.com/detachhead/basedpyright.git", branch: "main"
+
+  depends_on "node"
+
+  def install
+    system "npm", "install", *std_npm_args
+    bin.install_symlink libexec/"bin/pyright" => "basedpyright"
+    bin.install_symlink libexec/"bin/pyright-langserver" => "basedpyright-langserver"
+  end
+
+  test do
+    (testpath/"broken.py").write <<~EOS
+      def wrong_types(a: int, b: int) -> str:
+          return a + b
+    EOS
+    output = pipe_output("#{bin}/basedpyright broken.py 2>&1")
+    assert_match "error: Expression of type \"int\" is incompatible with return type \"str\"", output
```

Comment:
minor nit (ik this was actually from `pyright`)
```suggestion
    assert_match 'error: Expression of type "int" is incompatible with return type "str"', output
```